### PR TITLE
BC-6643 add data testid after vue3 upgrade

### PR DESCRIPTION
**Pull request by UzaeirKhan**
# Short Description

Adds `data-testids` in vue pages after vue 3 upgrade.

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-6643

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/feature-board/board/BoardAddCardButton.vue [hunk 1]
```diff
@@ -13,7 +13,9 @@
 			v-if="!isEditMode"
 		>
 			<VIcon>{{ mdiPlus }}</VIcon>
-			<span class="d-sr-only">{{ $t("components.board.action.addCard") }}</span>
+			<span class="d-sr-only" data-testid="add-card-1">{{
```
Comment (NFriedo) [inline src/components/feature-board/board/BoardAddCardButton.vue:16]: Just to be sure, it should be `add-card-1` and not `add-card`?

File: src/components/feature-board/card/CardTitle.vue [hunk 2]
```diff
@@ -14,6 +14,7 @@
 			:isFocused="isFocused"
 			@update:value="onUpdateValue"
 			@enter="onEnter"
+			data-testid="column-title"
```
Comment (NFriedo) [inline src/components/feature-board/card/CardTitle.vue:17]: This is the CardTitle component, so it should be probably `card-title` and not `column-title`. What do you think?

File: src/components/feature-board/board/BoardAddCardButton.vue (hunk 1 above)
Comment (UzaeirKhan) [reply src/components/feature-board/board/BoardAddCardButton.vue:16]: Agreed. Corrected.

File: src/components/feature-board/card/CardTitle.vue (hunk 2 above)
Comment (UzaeirKhan) [reply src/components/feature-board/card/CardTitle.vue:17]: You are correct. Changed to `card-title`

File: src/components/ui-board/BoardMenuAction.vue [hunk 3]
```diff
@@ -19,6 +19,7 @@ export default defineComponent({
 	name: "BoardMenuAction",
 	props: {
 		icon: { type: String as PropType<IconProps["icon"]> },
+		dataTestId: { type: String, default: "" },
```
Comment (NFriedo) [inline src/components/ui-board/BoardMenuAction.vue:22]: same as above in DefaultWireframe.vue :)

File: src/components/ui-board/BoardMenuActionDelete.vue [hunk 4]
```diff
@@ -2,7 +2,7 @@
 	<BoardMenuAction
 		:icon="mdiTrashCanOutline"
 		@click="onClick"
-		data-testid="board-menu-action-delete"
+		dataTestId="board-menu-action-delete"
```
Comment (NFriedo) [inline src/components/ui-board/BoardMenuActionDelete.vue:5]: I think we shouldn't change it to `dataTestId`. 

We decided that we want to use `data-testid` : https://documentation.dbildungscloud.dev/docs/nuxt-client/CodeConventions#data-testids

File: src/components/ui-board/BoardMenuActionEdit.vue [hunk 5]
```diff
@@ -2,7 +2,7 @@
 	<BoardMenuAction
 		:icon="mdiPencilOutline"
 		@click="onClick"
-		data-testid="board-menu-action-edit"
+		dataTestId="board-menu-action-edit"
```
Comment (NFriedo) [inline src/components/ui-board/BoardMenuActionEdit.vue:5]: see above, use `data-testid`

File: src/pages/administration/ClassOverview.page.vue [hunk 6]
```diff
@@ -3,7 +3,7 @@
 		:headline="t('pages.administration.classes.index.title')"
 		:breadcrumbs="breadcrumbs"
 		:full-width="true"
-		data-testid="admin-class-title"
+		dataTestId="admin-class-title"
```
Comment (NFriedo) [inline src/pages/administration/ClassOverview.page.vue:6]: see above, we want to use `data-testid` :)

File: src/components/templates/DefaultWireframe.vue [hunk 7]
```diff
@@ -95,6 +95,10 @@ export default defineComponent({
 			required: false,
 			default: false,
 		},
+		dataTestId: {
+			type: String,
+			default: "",
```
[the remark this answers is not in the document]
Comment (UzaeirKhan) [reply src/components/templates/DefaultWireframe.vue:100]: Corrected

File: src/components/ui-board/BoardMenuAction.vue (hunk 3 above)
Comment (UzaeirKhan) [reply src/components/ui-board/BoardMenuAction.vue:22]: Corrected

File: src/components/ui-board/BoardMenuActionDelete.vue (hunk 4 above)
Comment (UzaeirKhan) [reply src/components/ui-board/BoardMenuActionDelete.vue]: Corrected

File: src/components/ui-board/BoardMenuActionEdit.vue (hunk 5 above)
Comment (UzaeirKhan) [reply src/components/ui-board/BoardMenuActionEdit.vue]: Corrected

File: src/pages/administration/ClassOverview.page.vue (hunk 6 above)
Comment (UzaeirKhan) [reply src/pages/administration/ClassOverview.page.vue]: Corrected

**Review by NFriedo — APPROVED** [review]

**Pull request merged**